### PR TITLE
feat: add LocalDateTimeToInstantConverter (#16415)

### LINKED ISSUE #16415: feat: add LocalDateTimeToInstantConverter
### Describe your motivation

There already is a `LocalDateTimeToDateConverter` but `Instant` is a newer, better replacement. Since your Converter already relies on Instant to calculate the Date I would appreciate it, if `Instant` is also suportted.

### Describe the solution you'd like

Just add another Converter pretty simmilar to the `LocalDateTimeToDateConverter` without converting from Instant to Date.

### Describe alternatives you've considered

Since Converting from LocalDateTime to Instant is pretty easy, it could also be a oneliner like
`...
 binder.forField(startDateTimePicker)
                .withConverter(<LAMDBA HERE>,<AND HERE>)`

### Additional context

I would also provide a PR...

### PR DESCRIPTION
**Pull request by martinjunker**
## Description

There already is a `LocalDateTimeToDateConverter` but Instant is a newer, better replacement. Since your Converter already relies on `Instant` to calculate the `Date` I would appreciate it, if Instant is also supported.

Fixes #16415

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-data/src/main/java/com/vaadin/flow/data/converter/LocalDateTimeToInstantConverter.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.Objects;
+
+import com.vaadin.flow.data.binder.Result;
+import com.vaadin.flow.data.binder.ValueContext;
+
+/**
+ * A converter that converts between <code>LocalDateTime</code> and
+ * <code>Instant</code>.
+ *
+ * @author martinjunker
```
Comment (czp13) [inline flow-data/src/main/java/com/vaadin/flow/data/converter/LocalDateTimeToInstantConverter.java:30]:
```suggestion
```

File: flow-data/src/main/java/com/vaadin/flow/data/converter/LocalDateTimeToInstantConverter.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * @since 24.1
```
Comment (czp13) [inline flow-data/src/main/java/com/vaadin/flow/data/converter/LocalDateTimeToInstantConverter.java:31]:
```suggestion
```

**Comment by czp13** [comment]
We are lucky, all tests are passing after re executed them, most important bits to fix before able to merge are:
- adding covering tests,
- formatting files,
More info in my review, how to add tests or format the file. 🙇

**Comment by tepi** [comment]
Converter tests are probably covered by `com.vaadin.flow.data.converter.DefaultConverterFactoryTest` nowadays.

**Comment by mcollovati** [comment]
> Converter tests are probably covered by com.vaadin.flow.data.converter.DefaultConverterFactoryTest nowadays.

`DefaultConverterFactoryTest` does not test the single converter logic.
It would be nice to get the test back from V8.
It may be worth it to create a new ticket

**Comment by tepi** [comment]
> `DefaultConverterFactoryTest` does not test the single converter logic.

Ok, looks like it only tests that the types are correct. So, we are missing tests for the conversion logic of all the built-in converters, right? Definitely a separate ticket about that is needed.

File: flow-data/src/test/java/com/vaadin/flow/data/converter/LocalDateTimeToInstantConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        assertEquals(Result.ok(INSTANT_EXAMPLE),
+                converter.convertToModel(LOCAL_DATE_TIME_EXAMPLE, null));
+    }
+
+    @Test
+    public void testConvertToPresentation() {
+        assertEquals(LOCAL_DATE_TIME_EXAMPLE,
+                converter.convertToPresentation(INSTANT_EXAMPLE, null));
+    }
+}
```
Comment (czp13) [inline flow-data/src/test/java/com/vaadin/flow/data/converter/LocalDateTimeToInstantConverterTest.java:44]:
```suggestion
}

```

**Review by czp13 — APPROVED** [review]
You fixed everything discussed here. Thank you so much @martinjunker! 🙇

**Comment by martinjunker** [comment]
nice. thank you all!

**Pull request merged**